modcc/module.cpp

```diff
@@ -40,6 +40,15 @@ Module::Module(std::vector<char> const& buffer)
         buffer_.push_back(0);
 }
 
+Module::Module(const char* buffer, size_t count)
+{
```

Comment:
If you want to rangify it, go for it! We are all in the same source tree now.

On the performance side, calculating the size of the buffer to be inserted before inserting means that we have fewer `malloc` than with `push_back` one by one.